src/main/java/frc/robot/Robot.java

```diff
@@ -161,14 +165,27 @@ public void autonomousSetup() {
     m_autoSelected = m_chooser.getSelected();
     // m_autoSelected = SmartDashboard.getString("Auto Selector", kDefaultAuto);
     System.out.println("Auto selected: " + m_autoSelected);
+
+    auto = new Autonomous((DriveImpl) this.drive, (BallPathImpl) this.ballSubsystem);
```

Comment:
Once the constructor signature of `Autonomous` is changed to accept the interface types instead of the classes, then these type casts should be removed.